typing/typeclass.ml

```diff
@@ -1449,16 +1445,16 @@ let temp_abbrev loc env id arity uid =
        type_unboxed_default = false;
        type_uid = uid;
       }
-      env
   in
-  (!params, ty, env)
+  (!params, ty, ty_td)
 
 let initial_env define_class approx
-    (res, env) (cl, id, ty_id, obj_id, cl_id, uid) =
+    (res, env) (cl, id, ty_id, obj_id, uid) =
   (* Temporary abbreviations *)
   let arity = List.length cl.pci_params in
-  let (obj_params, obj_ty, env) = temp_abbrev cl.pci_loc env obj_id arity uid in
-  let (cl_params, cl_ty, env) = temp_abbrev cl.pci_loc env cl_id arity uid in
+  let (obj_params, obj_ty, obj_td) = temp_abbrev cl.pci_loc arity uid in
+  let env = Env.add_type ~check:true obj_id obj_td env in
+  let (cl_params, cl_ty, cl_td) = temp_abbrev cl.pci_loc arity uid in
```

Comment:
I don't understand why you needed to change this code here to add `obj_td` to the environment, if it was not needed before. Can you explain? (Isn't there a risk that this "temporary abbreviation" remains (shadowed) in the final environment? We have been seeing `#c/1` and `#c/2` in some error messages and I wonder if it is related.)

Reply:
Previously, `temp_abbrev` didn't return `cl_td`, which is needed for dummy `clty_hash_type`. Plus, do not need to treat `#t` type explicitly, we don't need to add dummy hash type to `env` anymore; so we decided not to modify `env` from `temp_abbrev`.

As a result, `obj_` related manipulations are intact (work as before), so it shouldn't be any issue.

Reply:
Ah, I missed this. It's not that you are adding one item to the env, it is that before you were adding one (inside the function) and now you are adding only one. Thanks for the clarification!